tests/ert/unit_tests/config/parsing/test_lark_parser.py

```diff
@@ -345,3 +345,40 @@ def test_that_giving_too_few_arguments_to_enkf_alpha_raises_config_validation_er
             file_name="config.ert",
             schema=init_user_config_schema(),
         )
+
+
+def test_that_mismatched_quotes_raises_config_validation_error():
+    with pytest.raises(ConfigValidationError, match='Did not expect character: "'):
+        _ = parse_contents(
+            """
+            NUM_REALIZATIONS 1
+            FORWARD_MODEL poly_eval(<FOO>=""bar")
+            """,
+            file_name="config.ert",
+            schema=init_user_config_schema(),
+        )
+
+
+def test_that_quotes_can_be_escaped():
+    contents = parse_contents(
+        """
+        NUM_REALIZATIONS 1
+        FORWARD_MODEL poly_eval(<FOO>="\\"bar")
+        """,
+        file_name="config.ert",
+        schema=init_user_config_schema(),
+    )
+    assert contents["FORWARD_MODEL"] == [["poly_eval", [("<FOO>", '\\"bar')]]]
```

Comment:
Can this be called "escaping"? The backslash character(s) is carried on to the forward model step as is, and not treating the quote character.

Reply:
It is escape in the sense that it doesn't end the string in our config. I think keeping the backslash is the correct behavior in order to preserve character locations and all information.

Reply:
👍🏻  Lets keep it that way until a user enters this narrow corner and proposes a valid use-case..